templates/templates/docs/openstack/docs.html

```diff
@@ -0,0 +1,149 @@
+                </a>
+              </h3>
+              {% else %}
+                <h3>{{ nav_group.navlink_text }}</h3>
+              {% endif %}
+            {% endif %}
+            {{ create_navigation(nav_group.children) }}
+          {% endfor %}
+        </div>
+      </div>
+    </aside>
+
+    <main class="col-9">
+      {{ document.body_html | safe }}
+
+			<div class="p-strip is-shallow">
+				<div class="p-notification--information">
+					<p class="p-notification__response">
+						Last updated {{ document.updated }}. <a href="{{ forum_url }}{{ document.topic_path }}">Help improve this document in the forum</a>.
+					</p>
+				</div>
+			</div>
+    </main>
+  </div>
+</div>
+
+<script>
+  var links = [].slice.call(document.querySelectorAll('.p-side-navigation--raw-html li > a'));
+  var currentPath = window.location.pathname;
+
+  links.forEach(function (link) {
+    link.addEventListener('click', function () {
+      links.forEach(function (link) {
+        link.removeAttribute('aria-current');
+      });
+      this.setAttribute('aria-current', 'page');
+      this.blur();
+    });
+
+    if (link.getAttribute('href') === currentPath) {
+      link.setAttribute('aria-current', 'page');
+    }
+  });
+
+  /**
+    Toggles the expanded/collapsed classed on side navigation element.
+
+    @param {HTMLElement} sideNavigation The side navigation element.
+    @param {Boolean} show Whether to show or hide the drawer.
+  */
+  function toggleDrawer(sideNavigation, show) {
+    if (sideNavigation) {
+      if (show) {
+        sideNavigation.classList.remove('is-collapsed');
+        sideNavigation.classList.add('is-expanded');
+      } else {
+        sideNavigation.classList.remove('is-expanded');
+        sideNavigation.classList.add('is-collapsed');
+      }
+    }
+  }
+
+  /**
+    Attaches event listeners for the side navigation toggles
+    @param {HTMLElement} sideNavigation The side navigation element.
+  */
+  function setupSideNavigation(sideNavigation) {
+    var toggles = [].slice.call(sideNavigation.querySelectorAll('.js-drawer-toggle'));
+
+    toggles.forEach(function (toggle) {
+      toggle.addEventListener('click', function (event) {
+        event.preventDefault();
+        var sideNav = document.getElementById(toggle.getAttribute('aria-controls'));
+
+        if (sideNav) {
+          toggleDrawer(sideNav, !sideNav.classList.contains('is-expanded'));
+        }
+      });
+    });
+  }
+
+  /**
+    Attaches event listeners for all the side navigations in the document.
+    @param {String} sideNavigationSelector The CSS selector matching side navigation elements.
+  */
+  function setupSideNavigations(sideNavigationSelector) {
+    // Setup all side navigations on the page.
+    var sideNavigations = [].slice.call(document.querySelectorAll(sideNavigationSelector));
+
+    sideNavigations.forEach(setupSideNavigation);
+  }
+
+  setupSideNavigations('.p-side-navigation, [class*="p-side-navigation--"]');
+</script>
```

Comment:
You've only used a shared include for the one tiny piece, of the `_search-box.html`, but basically the whole of this template is identical to the other 3 isn't it? You've used an include for the only piece that's different!

I would have thought that instead of creating `templates/docs/shared/_search-box.html`, you want to create `templates/shared/docs/_document.html` and put all of this content in there, including the search box.

```suggestion
  {% with 
    search_action="/openstack/docs/search",
    siteSearch="https://ubuntu.com/openstack/docs",
    placeholder="Search on Charmed Openstack Docs"
  %}
    {% include "shared/docs/_document.html" %}
  {% endwith %}
```